### ISSUE TITLE
Validation set

### ISSUE THREAD

**Issue by pengshancai** [CannyLab/summary_loop#9]
Does the train_summary_loop.py contain a validation step? (does not look like so)

**Comment by tingofurro** [comment]
Hey Pengshan,

Indeed the train_summary_loop script does not have validation.
Because we are training without supervision (references to overfit to), validation didn't seem required to us.

More concretely, we trained the model using a large set of news article, and the training procedure never went beyond the first epoch before we terminated, meaning that overfitting would be unlikely.

If you are using a smaller set of documents as the basis of training (on which you could overfit), or are seeing some overfitting then it'd be interesting to incorporate a validation step.

I'd love to hear your thoughts,

Philippe
